requester/requester.go

```diff
@@ -57,21 +64,16 @@ func New(cfg config.Config) *Requester {
 
 // Run starts the benchmark test and pipelines the results inside a Report.
 // Returns the Report when the test ended and all results have been collected.
-func (r *Requester) Run() (Report, error) {
-	req, err := r.config.HTTPRequest()
-	if err != nil {
-		return Report{}, fmt.Errorf("%w: %s", ErrRequest, err)
-	}
-
+func (r *Requester) Run(req *http.Request) (Report, error) {
```

Comment:
Won't having the same request passed by reference for each dispatch iteration be a problem?
We can easily copy the request value by giving `Requester` access to `config.Global.HTTPRequest()`. We can add this method to `request.Config` interface possibly.

Reply:
`Clone` also works ! Good to me 👌